src/core/ComboBox/ComboBox.tsx

```diff
@@ -180,189 +174,274 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     }
   }, [isOpen]);
 
+  // Initialize filtered options to the latest value options
   const [filteredOptions, setFilteredOptions] = React.useState(options);
   React.useEffect(() => {
     setFilteredOptions(options);
   }, [options]);
 
-  const [focusedIndex, setFocusedIndex] = React.useState(() =>
-    options.findIndex((option) => value === option.value),
-  );
-
-  // Maintain internal selected value state synced with `value` prop
-  const [selectedValue, setSelectedValue] = React.useState<T | undefined>(
-    value,
-  );
-  React.useEffect(() => {
-    setSelectedValue(value);
-  }, [value]);
-
-  // Controlled input value
-  const [inputValue, setInputValue] = React.useState<string>('');
-  const onInput = React.useCallback(
+  // Filter options based on input value
+  const [inputValue, setInputValue] = React.useState(inputProps?.value ?? '');
+  const handleOnInput = React.useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
-      setInputValue(event.target.value);
+      const { value } = event.currentTarget;
+      setInputValue(value);
+      dispatch(['open']);
+      setFilteredOptions(
+        filterFunction?.(options, value) ??
+          options.filter((option) =>
+            option.label.toLowerCase().includes(value.toLowerCase()),
+          ),
+      );
       inputProps?.onChange?.(event);
     },
-    [inputProps],
+    [filterFunction, inputProps, options],
   );
 
-  // update inputValue and focusedIndex every time selected value changes
+  // Reset focused item when filter changes
   React.useEffect(() => {
```

Comment:
Can it be that this reset is causing you a bug where initialy selected item is not focused?

Reply:
I tried to remove it and it didn't fix the bug